src/VirtoCommerce.Platform.Web/Controllers/Api/PlatformExportImportController.cs

```diff
@@ -188,21 +195,19 @@ public ActionResult Cancel([FromRoute] string jobId)
         [HttpGet]
         [Route("export/download/{fileName}")]
         [Authorize(Permissions.PlatformExport)]
-        public ActionResult DownloadExportFile([FromRoute] string fileName)
+        public async Task<ActionResult> DownloadExportFile([FromRoute] string fileName)
         {
-            var localTmpFolder = Path.GetFullPath(Path.Combine(_platformOptions.DefaultExportFolder));
-            var localPath = Path.Combine(localTmpFolder, Path.GetFileName(fileName));
+            var localPath = Path.Combine(_platformOptions.DefaultExportFolder, Path.GetFileName(fileName));
 
-            //Load source data only from local file system
-            using (var stream = System.IO.File.Open(localPath, FileMode.Open))
+            var provider = new FileExtensionContentTypeProvider();
+            if (!provider.TryGetContentType(localPath, out var contentType))
             {
-                var provider = new FileExtensionContentTypeProvider();
-                if (!provider.TryGetContentType(localPath, out var contentType))
-                {
-                    contentType = "application/octet-stream";
-                }
-                return PhysicalFile(localPath, contentType);
+                contentType = "application/octet-stream";
             }
+
+            using var blobStream = await _blobStorageProvider.OpenReadAsync(_blobUrlResolver.GetAbsoluteUrl(localPath));
+            var blobBytes = blobStream.ReadFully();
```

Comment:
Why can't the stream be returned instead of being read into memory?   this can leads to high memory utilization
`File(blobStream, contentType)`